plugins/pulp_docker/plugins/distributors/publish_steps.py

```diff
@@ -220,3 +223,154 @@ def process_main(self):
+                                    against RepoContentUnit model
+        :type  repo_content_unit_q: mongoengine.Q
+        :param repo: The repo being worked on
+        :type  repo: pulp.plugins.model.Repository
+        :param config: The publish configuration
+        :type  config: PluginCallConfiguration
+        :param remote_repo_path: relative path on remote server where published repo should live
+        :type remote_repo_path: str
+        """
+
+        super(PublishTagsForRsyncStep, self).__init__(step_type, [models.Tag],
+                                                      repo_content_unit_q=repo_content_unit_q,
+                                                      repo=repo, config=config,
+                                                      remote_repo_path=remote_repo_path,
+                                                      published_unit_path=['manifests'])
+        self._tag_names = set()
+        self.repo_registry_id = repo_registry_id
+
+    def process_main(self, item=None):
+        """
+        Create the manifest tag relative links.
+
+        :param item: The tag to process
+        :type  item: pulp_docker.plugins.models.Tag
+        """
+        manifest = models.Manifest.objects.get(digest=item.manifest_digest)
+        filename = item.name
+        symlink = self.make_link_unit(manifest, filename, self.get_working_dir(),
+                                      self.remote_repo_path,
+                                      self.get_config().get("remote")["root"],
+                                      self.published_unit_path)
+        self.parent.symlink_list.append(symlink)
+        self._tag_names.add(item.name)
+
+    def finalize(self):
+        """
+        Write the Tag list file so that clients can retrieve the list of available Tags.
+        """
+        tags_path = os.path.join(self.parent.get_working_dir(), '.relative', 'tags')
+        misc.mkdir(tags_path)
+        with open(os.path.join(tags_path, 'list'), 'w') as list_file:
+            tag_data = {
+                'name': self.repo_registry_id,
+                'tags': list(self._tag_names)}
+            list_file.write(json.dumps(tag_data))
+        # We don't need the tag names anymore
+        del self._tag_names
+
+
+class DockerRsyncPublisher(Publisher):
+
+    REPO_CONTENT_TYPES = [constants.IMAGE_TYPE_ID, constants.BLOB_TYPE_ID,
+                          constants.MANIFEST_TYPE_ID]
+
+    REPO_CONTENT_MODELS = [models.Blob, models.Manifest, models.Image]
+
+    def _get_predistributor(self):
+        """
+        Returns the distributor object representing the predistributor. A predistributor is the
+        distributor that is used to publish a repository prior to publishing with the rsync
+        distributor.
+
+        :return: predistributor that was configured in rsync distributor's config
+        :rtype: Distributor
+        """
+        return None
```

Comment:
Is this method actually required? I suspect that these steps at runtime won't ever call this method which would allow the base implementation to be adequate.